Document MenuBar and ContextMenu (#7502)
slint-ui/slint: docs/astro/src/content/docs/reference/window/window.mdx

<SlintProperty propName="title" typeName="string">
The window title that is shown in the title bar.
</SlintProperty>

## `MenuBar` element

> A `Window` can contain one `MenuBar` element. This special element is used to create a menu bar for that window.
> There can only be one `MenuBar` element in a `Window` and it must not be in a `for` or a `if`.
> The `MenuBar` doesn't have properties, but it should contain <Link type="MenuItem" /> as children that represent top level entries in the menu bar.
> Each `MenuItem` can contain further `MenuItem` as children that represent sub-menu

Reviewer: ```suggestion
Place a single `MenuBar` element into a `Window` to declare a menu bar for the window.

:::note{Note}
There can only be one `MenuBar` element in a `Window` and it must not be in a `for` or a `if`.
:::

The `MenuBar` doesn't have properties, but it should contain <Link type="MenuItem" /> as children that represent top level entries in the menu bar.
Nest `MenuItem` elements to declare sub-menus.
```

Depending on the platform, the menu bar might be native or rendered by Slint.

> This means that for example, on MacOs, the menu bar be in top bar on the desktop where application usually have their top bar.
> The `width` and `height` property of the `Window` are the one of the client area, without counting the menu bar.
> The `x` and `y` properties of `Window` children are also relative to the client area.

Reviewer: (The "at top of the screen" phrase is the term used in the Apple HIG, and it's short and precise)

```suggestion
This means that for example, on macOS, the menu bar will be at the top of the screen.
The `width` and `height` property of the `Window` define the client area, excluding the menu bar.
The `x` and `y` properties of `Window` children are also relative to the client area.
```

### Example

```slint
export component Example inherits Window {

>     MenuBar {
>         MenuItem {
>             title: "File";
>             MenuItem {
>                 title: "New";
>             }
>             MenuItem {
>                 title: "Open";
>             }
>         }
>         MenuItem {
>             title: "Edit";
>             MenuItem {
>                 title: "Copy";
>             }
>             MenuItem {
>                 title: "Paste";
>             }
>         }
>     }
>     // ... actual window content goes here

Reviewer: Let's add a sub-menu, to show the readers how the nesting looks - and added `@tr`.

```suggestion
    MenuBar {
        MenuItem {
            title: @tr("File");
            MenuItem {
                title: @tr("New");
            }
            MenuItem {
                title: @tr("Open");
            }
        }
        MenuItem {
            title: @tr("Edit");
            MenuItem {
                title: @tr("Copy");
            }
            MenuItem {
                title: @tr("Paste");
            }
            MenuItem {
                title: @tr("Find");
                MenuItem {
                    title: @tr("Find in document...");
                }
                MenuItem {
                    title: @tr("Find Next");
                }
                MenuItem {
                    title: @tr("Find Previous");
                }
            }
        }
    }
    // ... actual window content goes here
```
